framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -34,5 +37,8 @@ MaterialStdVectorAux::~MaterialStdVectorAux()
 Real
 MaterialStdVectorAux::getRealValue()
 {
+  mooseAssert(_special_qp < _q_point.size(), "MaterialStdVectorAux.  special_qp specified as " << _special_qp << " but there are only " << _q_point.size() << " quadpoints in the element");
```

Comment:
Well, that is exactly what we want to do though, _indicate an invalid/error state without crashing_. It should indicate that no valid data can be provided for the elements in question because the QP does not exist on them.